runestone/activecode/test/test_activecode.py

```diff
@@ -1,242 +1,209 @@
-        try:
-            self.wait.until(EC.presence_of_element_located((By.ID, "test1")))
-        except TimeoutException:
-            text = self.driver.page_source
-            print(text[:300])
-        t1 = self.driver.find_element_by_id("test1")
-        self.assertIsNotNone(t1)
-        rb = t1.find_element_by_class_name("run-button")
-        self.assertIsNotNone(rb)
-        rb.click()
-        self.wait.until(
-            EC.text_to_be_present_in_element((By.ID, "test1_stdout"), "Hello World"),
-            message="Did not find expected text",
-        )
-        output = t1.find_element_by_class_name("ac_output")
-        self.assertEqual(output.text.strip(), "Hello World")
-
-    def test_hidden(self):
-        """
-        1. Get the outer div id of the activecode component
-        2. Find the run button using its class name
-        3. Run the example
-        4. Check the output from the ac_output element
-        :return:
-        """
-        self.driver.get(self.host + "/index.html")
-        try:
-            self.wait.until(EC.presence_of_element_located((By.ID, "testprefixcode")))
-        except TimeoutException:
-            text = self.driver.page_source
-            print(text[:300])
-        t1 = self.driver.find_element_by_id("testprefixcode")
-        self.assertIsNotNone(t1)
-        rb = t1.find_element_by_class_name("run-button")
-        self.assertIsNotNone(rb)
-        rb.click()
-        self.wait.until(
-            EC.text_to_be_present_in_element(
-                (By.ID, "testprefixcode_stdout"), "My Code"
-            )
-        )
-        output = t1.find_element_by_class_name("ac_output")
-        self.assertIn("hidden code", output.text.strip())
-        self.assertIn("i\nx", output.text.strip())
-
-    def test_history(self):
-        """
-        1. Get the outer div id of the activecode component
-        2. Find the run button using its class name
-        3. Run the example
-        4. Check the output from the ac_output element
-        :return:
-        """
-        self.driver.get(self.host + "/index.html")
-        t1 = self.driver.find_element_by_id("test1")
-        self.assertIsNotNone(t1)
-        rb = t1.find_element_by_class_name("run-button")
-        self.assertIsNotNone(rb)
-        try:
-            rb.click()
-        except ElementClickInterceptedException:
-            # interceptor = self.driver.find_element_by_class_name("navbar-collapse")
-            # self.driver.execute_script(
-            #     """
-            # var el = arguments[0];
-            # el.parentNode.removeChild(el);
-            # """,
-            #     interceptor,
-            # )
-            self.driver.execute_script("window.scrollTo(0, 0);")
-            rb.click()
-
-        ta = t1.find_element_by_class_name("cm-s-default")
-        self.assertIsNotNone(ta)
-        self.driver.execute_script(
-            """window.edList['test1'].editor.setValue("print('GoodBye')")"""
```

Comment:
@bnmnetp, I'm having trouble with this line when doing server-side testing with selectquestion. While the component runs as expected in a selectquestion, this statement doesn't update the text in the activecode window; it works fine here, though. Any ideas on another way to programmatically change the text in an activecode?

Reply:
Hmmm, What is the problem?  Is there a javascript error?

Could we use a method similar to what you use for fill in the blank?

Reply:
One way that should definitely work would be to send a `runlog` entry to the server to pre load history and then click on the load history button for the activecode.

But it is still not at all clear why the execute_script method is not working.

Reply:
The tricky part is getting text to CodeMirror. It  does lots of JS tricks that make the actual element that receives keystrokes hidden behind a bunch of divs that display line numbers, syntax highlighting, etc. So, the Selenium `send_keys` method doesn't work (to the best of my knowledge) since we can't find the correct HTML element to send it to.

In the client-side tests, the JS here works just fine. When run outside a selectquestion, it's also fine. But, evidently the way selectquestion loads the activecode makes this not work.

I was hoping you'd have some clue as to why -- this is pretty tied up inside selectquestion/activecode, and is pretty far from the parts of the code I know...

Reply:
Hmm, the `runlog` approach seems reasonable. It's fairly easy to debug -- just insert an `import pdb; pdb.set_trace()` just before the failure, then open the Chrome browser, open the debug pane, and play. Since the server is up and running, you can both do client-side playing and server-side playing.

...but, as I said before, I don't really know the system well enough to debug it...

Reply:
Having said that it does not look like it is failing?  

In any case I'm assuming you mean on the line right before execute_script?

Reply:
The tests fail only on the server -- I'll push the server tests. (They'll fail, since these changes aren't merged). These tests pass, but will probably need a bit of tweaking at this line to get the server-side tests fully working.

Reply:
Yes, exactly. Do you want to chat briefly via Zoom? I'm on Slack.

Reply:
Note that none of the tests currently fail, since I marked the failing server-side selectquestion test with a skip.